include/deal.II/base/mpi.h

```diff
@@ -704,6 +773,36 @@ namespace Utilities
        * MPI process.
        */
       ~MPI_InitFinalize();
+
+      /**
+       * Register a reference to an MPI_Request
+       * that we MPI_Wait on before calling MPI_Finalize.
+       *
+       * The object @p request needs to exist when MPI_Finalize is called, which means the
+       * request is typically statically allocated. Note that is is acceptable
+       * for a request to be already waited on (and consequently reset to
+       * MPI_REQUEST_NULL).
+       *
+       * A typical usage is the following idiom:
+       * @code
+       * void my_fancy_communication()
+       * {
+       *   static MPI_Request request = MPI_REQUEST_NULL;
+       *   MPI_InitFinalize::register_static_request(request);
+       *   MPI_Wait(&request, MPI_STATUS_IGNORE);
+       *   // [some algorithm that is not safe to be executed twice in a row.]
+       *   MPI_IBarrier(comm, &request);
+       * }
+       * @endcode
+       */
+      static void
+      register_static_request(MPI_Request &request);
```

Comment:
Ah, I see how you work around the issue of registration. The issue I see is that while you say that the request object needs to be `static`, you really don't have a way to enforce this, and will likely get very difficult to track down segfaults at the end of the program if someone calls the function with a non-`static` variable.